brickstrap.sh

```diff
@@ -19,29 +20,51 @@
 # http://twitter.com/kvz
 #
 
-### Configuration
+### Runtime
 #####################################################################
+set -e
+
+# Bash will remember & return the highest exitcode in a chain of pipes.
+set -o pipefail
 
 # Environment variables
 [ -z "${LOG_LEVEL}" ] && LOG_LEVEL="3" # 4 = debug -> 0 = fail
 
-# Commandline options. This defines the usage page
+#
+# Get the brickstrap source directory
+#
+function br_script_path()
+{
+    SCRIPT_PATH=$(readlink -f "$0")
+    SCRIPT_PATH=$(dirname "$SCRIPT_PATH")
+    echo -n "$SCRIPT_PATH"
+}
 
-read -r -d '' usage <<-'EOF'
-  usage: brickstrap [-b <board>] [-d <dir>] <command>
+. "$(br_script_path)/brickstrap-components.sh"
+
+# Commandline options. This defines the usage page
+BRP_USAGE=$(cat <<'EOF'
+Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
```

Comment:
I know we discussed earlier about saving things like which qemu in the rootfs, but looking over the commands here, I am seeing a need to keep this info outside of the rootfs.

First, we need to expand the usage description to show which options are valid with which commands. Here is what I propose

    Usage: brickstrap <options> <command>
    Creation:
    brickstrap -p <project> -c <component> [ -c <component> ... ] [ -d <rootfs> ] { create-conf | create-rootfs | all }
    Incremental build:
    brickstrap -d <rootfs> { run-multistrap | copy-root | configure-packages | run-hooks | create-tar | create-image | create-report }
    Debug:
    brickstrap -d <rootfs> { simulate-multistrap | run-hook <hook> | shell [ <shell> ] }
    Destruction:
    brickstrap -d <rootfs> delete